File: scripts/cf_fix.mjs
import https from 'https';

const TOKEN   = 'Tx4LEBXQzIORQEIOQfmCFghvtEQ1QeQO7szlCErx_y8.XQX53nLZc3wpglcPNSld3zKdH-_43Cc5X7LWYKLP2bw';
const ACCOUNT = 'fac7207421271dd5183fcab70164cad1';
const ZONE    = '01a040ee5bbab5c1c2a0e73ddd5ca57e';

function api(method, path, body) {
  return new Promise((resolve, reject) => {
    const data = body ? JSON.stringify(body) : undefined;
    const opts = {
      hostname: 'api.cloudflare.com',
      path: '/client/v4' + path,
      method,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
        'Content-Type': 'application/json',
        ...(data ? { 'Content-Length': Buffer.byteLength(data) } : {}),
      },
    };
    const req = https.request(opts, res => {
      let d = '';
      res.on('data', c => d += c);
      res.on('end', () => resolve(JSON.parse(d)));
    });
    req.on('error', reject);
    if (data) req.write(data);
    req.end();
  });
}

// 1. Delete Pages custom domain
console.log('=== 1. Delete Pages custom domain (remove SSL conflict) ===');
const del = await api('DELETE', `/accounts/${ACCOUNT}/pages/projects/neowow-studio-dashboard/domains/app.neowow.studio`);
console.log('  result:', del.success ? 'OK deleted' : JSON.stringify(del.errors || del));

// 2. List DNS records
console.log('\n=== 2. DNS records for app.neowow.studio ===');
const dnsResp = await api('GET', `/zones/${ZONE}/dns_records?name=app.neowow.studio`);
const recs = dnsResp.result || [];
recs.forEach(r => console.log(`  id=${r.id} type=${r.type} content=${r.content} proxied=${r.proxied}`));
if (!recs.length) console.log('  NO_RECORDS');

// 3. Ensure CNAME exists and is proxied
console.log('\n=== 3. Ensure CNAME app -> pages.dev (proxied=true) ===');
if (!recs.length) {
  const cr = await api('POST', `/zones/${ZONE}/dns_records`, {
    type: 'CNAME', name: 'app',
    content: 'neowow-studio-dashboard.pages.dev',
    proxied: true, ttl: 1,
  });
  console.log('  Created:', cr.success ? 'OK' : JSON.stringify(cr.errors));
} else {
  const rec = recs[0];
  if (!rec.proxied || rec.content !== 'neowow-studio-dashboard.pages.dev') {
    const pr = await api('PATCH', `/zones/${ZONE}/dns_records/${rec.id}`, {
      proxied: true, content: 'neowow-studio-dashboard.pages.dev',
    });
    console.log(`  Patched id=${rec.id}:`, pr.success ? 'OK' : JSON.stringify(pr.errors));
  } else {
    console.log('  Already correct (proxied=true, correct target)');
  }
}

// 4. SSL cert packs
console.log('\n=== 4. SSL certificate packs ===');
const ssl = await api('GET', `/zones/${ZONE}/ssl/certificate_packs`);
(ssl.result || []).forEach(p =>
  console.log(`  type=${p.type} status=${p.status} hosts=${JSON.stringify(p.hosts)}`)
);

// 5. Test connectivity
console.log('\n=== 5. Test HTTPS app.neowow.studio ===');
const test = await new Promise(resolve => {
  const req = https.request({ hostname: 'app.neowow.studio', path: '/', method: 'GET' }, res => {
    resolve({ status: res.statusCode, server: res.headers.server || '', cf: res.headers['cf-ray'] || '' });
  });
  req.on('error', e => resolve({ error: e.message }));
  req.end();
});
console.log('  result:', JSON.stringify(test));

File: scripts/cf_fix.py
import urllib.request, urllib.error, json, sys

TOKEN   = "Tx4LEBXQzIORQEIOQfmCFghvtEQ1QeQO7szlCErx_y8.XQX53nLZc3wpglcPNSld3zKdH-_43Cc5X7LWYKLP2bw"
ACCOUNT = "fac7207421271dd5183fcab70164cad1"
BASE    = "https://api.cloudflare.com/client/v4"
HEADERS = {"Authorization": f"Bearer {TOKEN}", "Content-Type": "application/json"}

def api(method, path, data=None):
    url  = BASE + path
    body = json.dumps(data).encode() if data else None
    req  = urllib.request.Request(url, data=body, headers=HEADERS, method=method)
    try:
        with urllib.request.urlopen(req) as r:
            return json.loads(r.read())
    except urllib.error.HTTPError as e:
        return json.loads(e.read())

# 1. Delete Pages custom domain
print("=== 1. Deleting Pages custom domain (remove Worker/Pages SSL conflict) ===")
r = api("DELETE", f"/accounts/{ACCOUNT}/pages/projects/neowow-studio-dashboard/domains/app.neowow.studio")
print(" success:", r.get("success"), r.get("errors", ""))

# 2. Get Zone ID
print("\n=== 2. Getting Zone ID ===")
r = api("GET", "/zones?name=neowow.studio")
if not r.get("result"):
    print("FAIL: zone not found"); sys.exit(1)
zone = r["result"][0]["id"]
print(f" zone={zone}")

# 3. List DNS records
print("\n=== 3. DNS records for app.neowow.studio ===")
r = api("GET", f"/zones/{zone}/dns_records?name=app.neowow.studio")
recs = r.get("result", [])
for rec in recs:
    print(f"  id={rec['id']} type={rec['type']} content={rec['content']} proxied={rec['proxied']}")
if not recs:
    print("  NO_RECORDS")

# 4. Ensure CNAME exists and is proxied
print("\n=== 4. Ensuring CNAME app -> pages.dev (proxied=true) ===")
if not recs:
    r = api("POST", f"/zones/{zone}/dns_records", {
        "type": "CNAME", "name": "app",
        "content": "neowow-studio-dashboard.pages.dev",
        "proxied": True, "ttl": 1
    })
    print(" Created:", "OK" if r.get("success") else r)
else:
    rec_id = recs[0]["id"]
    r = api("PATCH", f"/zones/{zone}/dns_records/{rec_id}", {
        "proxied": True,
        "content": "neowow-studio-dashboard.pages.dev"
    })
    print(f" Patched id={rec_id}:", "OK" if r.get("success") else r)

# 5. SSL cert status
print("\n=== 5. Universal SSL cert status ===")
r = api("GET", f"/zones/{zone}/ssl/certificate_packs")
for p in r.get("result", []):
    print(f"  type={p.get('type')} status={p.get('status')} hosts={p.get('hosts', [])}")
